### ISSUE TITLE
y_ture.shape is expected to be (N,3) yet (?,?,?,?) is received

### ISSUE THREAD

**Issue by ahmelshi** [fchollet/keras#7814]
My model looks something like this :

now the problem is in the y_ture.shape as whenever i try to access y_trrue to do needed calculation inside the loss function i get y_true.shape as 

y_true (?, ?, ?, ?)
y_true Tensor("decoder/FinalConv_target:0", shape=(?, ?, ?, ?), dtype=float32)

Thank you!

- [-] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [-] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [-] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [-] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
